### ISSUE TITLE
Adding atomic number attributes to topology readers

### ISSUE THREAD

**Issue by IAlibay** [MDAnalysis/mdanalysis#2758]
## Is your feature request related to a problem? ##

I'm raising this mostly in the context of the TOPParser, but it might be useful to consider in terms of future dealings with topology types that contain atomic numbers (TPR?).

Our current approach with the TOPParser is that we automatically convert atomic number records to elements. However, with PARM7 files, there are cases where the atomic number can either be wrong (e.g. ff03ua https://github.com/MDAnalysis/mdanalysis/issues/2651#issuecomment-605658083) or given a dummy value (0 or -1).

As mentioned by @lilyminium (https://github.com/MDAnalysis/mdanalysis/issues/2651#issuecomment-605711880), it might be useful to store atomic numbers so that users can do their own topology attribute manipulations post-reading.

## Possible solutions ##

As far as I am aware the only topology parser to deal with atomic numbers is DMS and its `atomnum` attribute.

If we want to keep atomic numbers, we could go down two possible routes:

1) We just store the atomic number record as-is from the topology file and also do the atomic number -> elements conversion.
2) We only store what is in the file (i.e. atomic record) and consider the atomic number -> elements conversion something that users should be doing themselves.

An additional question here is, what would we like to call an atomic number topology attribute? As mentioned, `atonum` is currently a thing but @lilyminium raised some concerns about its use here: https://github.com/MDAnalysis/mdanalysis/pull/2714#discussion_r436305351

## Describe alternatives you've considered ##

Alternatively, we don't store atomic number records at all.

## Additional context ##

This discussion thread might be relevant here: https://github.com/MDAnalysis/mdanalysis/pull/2714#discussion_r436305351

**Comment by IAlibay** [comment]
Tagging @cbouy here, whom I believe mentioned at some point that atomic numbers may be required for MDA -> RDKIT things (please do correct me if I'm wrong).

**Comment by cbouy** [comment]
> Tagging @cbouy here, whom I believe mentioned at some point that atomic numbers may be required for MDA -> RDKIT things (please do correct me if I'm wrong).

Yes, although it's not strictly required, because RDKit can create Atoms from the atomic number or the symbol. But under the hood it converts the symbol to the corresponding atomic number in their periodic table, and then constructs the Atom. So I believe it would be nicer to have an atomic number attribute if it's available in the original file.
